ERCS/erc-7969.md

```diff
@@ -0,0 +1,261 @@
+                   "KBgQDwIRP/UC3SBsEmGqZ9ZJW3/DkMoGeLnQg1fWn7/zYt"
+                   "IxN2SnFCjxOCKG9v3b4jYfcTNh5ijSsq631uBItLa7od+v"
+                   "/RtdC2UzJ1lWT947qR+Rcac2gbto/NMqJ0fzfVjH4OuKhi"
+                   "tdY9tf6mcwGjaNBcWToIMmPSPDdQPNUYckcQ2QIDAQAB")
+                   ```
+   ````
+
+   The process to register this key would be:
+
+   - 1. Extract the public key value from the p= parameter:
+
+   ```
+   MIGfMA0GCSqGSIb3DQEBAQUAA4GNADCBiQKBgQDwIRP/UC3SBsEmGqZ9ZJW3/DkMoGeLnQg1fWn7/zYtIxN2SnFCjxOCKG9v3b4jYfcTNh5ijSsq631uBItLa7od+v/RtdC2UzJ1lWT947qR+Rcac2gbto/NMqJ0fzfVjH4OuKhitdY9tf6mcwGjaNBcWToIMmPSPDdQPNUYckcQ2QIDAQAB
+   ```
+
+   - 2. Calculate the keccak256 hash of the public key:
+
+   ```solidity
+   bytes32 keyHash = keccak256(bytes("MIGfMA0GCSqGSIb3DQEBAQUAA4GNADCBiQKBgQDwIRP/UC3SBsEmGqZ9ZJW3/DkMoGeLnQg1fWn7/zYtIxN2SnFCjxOCKG9v3b4jYfcTNh5ijSsq631uBItLa7od+v/RtdC2UzJ1lWT947qR+Rcac2gbto/NMqJ0fzfVjH4OuKhitdY9tf6mcwGjaNBcWToIMmPSPDdQPNUYckcQ2QIDAQAB"));
+   ```
+
+   - 3. Calculate the domain hash:
+
+   ```solidity
+   bytes32 domainHash = keccak256(bytes("example.org"));
+   ```
+
+   - 4. Register the key hash in the registry:
+
+   ```solidity
+   registry.setKeyHash(domainHash, keyHash);
+   ```
+
+### Events
+
+#### KeyHashRegistered
+
+This event MUST be emitted when a new DKIM public key hash is registered for a domain.
+
+```solidity
+event KeyHashRegistered(bytes32 domainHash, bytes32 keyHash)
+```
+
+#### KeyHashRevoked
+
+This event MUST be emitted when a DKIM public key hash is revoked for a domain.
+
+```solidity
+event KeyHashRevoked(bytes32 domainHash)
+```
+
+### Functions
+
+#### isKeyHashValid
+
+This function MUST return `true` if the provided key hash is valid for the given domain hash, and `false` otherwise.
+
+```solidity
+function isKeyHashValid(
+    bytes32 domainHash,
+    bytes32 keyHash
+) external view returns (bool)
+```
+
+## Rationale
+
+The interface is designed to be simple and focused on the core functionality of DKIM public key hash registration and validation. The use of keccak256 hashing for both domain names and public keys ensures consistent and secure handling of the data.
+
+The events allow for efficient tracking of key registrations and revocations, which is important for maintaining the integrity of the registry.
+
+## Backwards Compatibility
+
+This EIP introduces a new interface and does not affect existing contracts or standards.
+
+## Security Considerations
+
+1. Domain owners must ensure they have control over their private keys and domain names.
+2. The registry implementation should include proper access control mechanisms to prevent unauthorized registrations.
+3. The registry should implement a mechanism to handle key rotation and revocation.
+4. Implementations should consider rate limiting to prevent spam registrations.
+
+## Reference Implementation
+
+```solidity
+// SPDX-License-Identifier: MIT
```

Comment:
Same CC0-1.0 comment.

Reply:
Thank you for your comment. I directly removed the license as other ERCs usually don't have it as well.